### PR TITLE
feat(backend): onboarding questions and answers bb-133

### PR DESCRIPTION
**Pull request by ihorLysak**
### Summary
Created migration files that add `onboarding_questions` and `onboarding_answers` tables to the database, created migrations that populate named tables with data.

### CODE REVIEW

File: apps/backend/src/db/migrations/20240823121159_add_onboarding_questions_data.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+} as const;
+
+const QUESTIONS = [
+	"Can you tell me a bit about yourself? What’s currently important to you in your life?",
+	"What are your key roles in life right now?",
+	"Do you have a partner?",
+	"Which areas do you feel most challenged by right now?",
+	"What specific outcomes would you like to see by using the app?",
+	"Are there any specific types of tasks or practices you enjoy or find particularly motivating?",
+	"Spiritual",
```
Comment (ihorLysak) [inline apps/backend/src/db/migrations/20240823121159_add_onboarding_questions_data.ts:16]: This sould not be here

File: apps/backend/src/db/migrations/20240823115158_create_onboarding_questions_table.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { type Knex } from "knex";
+
+const TABLE_NAME = "onboarding_questions";
+
+const ColumnName = {
+	CREATED_AT: "created_at",
+	ID: "id",
+	QUESTION: "question",
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823115158_create_onboarding_questions_table.ts:8]:
```suggestion
	LABEL: "label",
```
Let's have the same as with quiz questions

File: apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+async function up(knex: Knex): Promise<void> {
+	await knex.schema.createTable(TABLE_NAME.ONBOARDING_ANSWERS, (table) => {
+		table.increments(ColumnName.ID).primary();
+		table.string(ColumnName.ANSWER).unique().notNullable();
+		table
+			.integer(ColumnName.QUESTION_ID)
+			.references(ColumnName.ID)
+			.inTable(TABLE_NAME.ONBOARDING_QUESTIONS)
+			.notNullable();
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts:24]: Add delete strategy CASCADE

File: apps/backend/src/db/migrations/20240823125503_add_onboarding_answers_data.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		const questions: { id: number }[] = await transaction(
+			TABLE_NAME.ONBOARDING_QUESTIONS,
+		).select("id");
+
+		const answerInserts = questions.flatMap((question, index) => {
+			return ANSWERS[index]?.map((answer) => ({
+				answer,
+				question_id: question.id,
+			}));
+		});
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823125503_add_onboarding_answers_data.ts:61]: Not a good idea. It's better to have a structure like {"question": [...answers] }
Check migrations for quiz questions

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

File: apps/backend/src/db/migrations/20240823115158_create_onboarding_questions_table.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { type Knex } from "knex";
+
+const TableName = "onboarding_questions";
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823115158_create_onboarding_questions_table.ts:3]:
```suggestion
const TABLE_NAME = "onboarding_questions";
```

File: apps/backend/src/db/migrations/20240823121159_add_onboarding_questions_data.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { type Knex } from "knex";
+
+const TableName = "onboarding_questions";
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823121159_add_onboarding_questions_data.ts:3]:
```suggestion
const TABLE_NAME = "onboarding_questions";
```

File: apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+async function up(knex: Knex): Promise<void> {
+	await knex.schema.createTable(TableName.ONBOARDING_ANSWERS, (table) => {
+		table.increments(ColumnName.ID).primary();
+		table.string(ColumnName.LABEL).unique().notNullable();
+		table
+			.integer(ColumnName.QUESTION_ID)
+			.references(ColumnName.ID)
+			.inTable(TableName.ONBOARDING_QUESTIONS)
+			.onDelete("CASCADE")
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts:24]:
```suggestion
			.onDelete(DELETE_STRATEGY)
```
Move to a separate variable

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import { type Knex } from "knex";
+
+const TABLE_NAME = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240823123910_create_onboarding_answers_table.ts:3]: Check out the rules B22 and B23

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**